src/bin/ion/commands/stats.rs

```diff
@@ -117,34 +148,38 @@ fn top_level_max_depth(value: LazyValue<AnyEncoding>) -> usize {
     let mut stack = vec![(value, 0)];
     while let Some((current_value, depth)) = stack.pop() {
         max_depth = max(max_depth, depth);
-        if current_value.ion_type() == IonType::Struct {
-            let struct_value = current_value.read().unwrap().expect_struct().unwrap();
-            for field in struct_value {
-                stack.push((field.unwrap().value(), depth + 1));
+        use ValueRef::*;
+        match current_value.read().unwrap() {
+            Struct(s) => {
+                for field in s {
+                    stack.push((field.unwrap().value(), depth + 1));
```

Comment:
It would probably be good to return an `anyhow::Result` instead of `.unwrap()`ping  all of the errors in this method.